lib/src/Pos/Diffusion/Full.hs

```diff
@@ -351,10 +354,13 @@ runDiffusionLayerFull networkConfig transport ourVerInfo mEkgNodeMetrics oq keep
         let itMap = M.fromList itList
         return ((>>= either throwM return) <$> itMap)
     subscriptionThread nc sactions = case topologySubscriptionWorker (ncTopology nc) of
-        Just (SubscriptionWorkerBehindNAT dnsDomains) ->
-            dnsSubscriptionWorker oq networkConfig dnsDomains keepaliveTimer slotDuration sactions
+        Just (SubscriptionWorkerBehindNAT dnsDomains) -> do
+            -- TODO wire this timer up so that we can reset it when a new block
+            -- arrives.
+            timer <- newDynamicTimer (convertUnit <$> slotDuration)
```

Comment:
Do we not need the timer to be in scope within the diffusion layer? If so, get rid of the `_keepaliveTimer` argument.

Reply:
Ah, but we *do* need it to be in scope, so that we can reset it when a new block arrives (as the comment says).